### PR TITLE
Add whisper-set-xfilesfactor.py 

### PR DESCRIPTION
**Pull request by piotr1212**
Add setXFilesFactor function which changes the XFilesFactor
without modifying the aggregation method.
Refactor setAggregationMethod, split out the writing of header
metadata.
Add validation for writing header medatadat.
test_whisper.py: force rereading header when simulating a
corrupt whisper file

### CODE REVIEW

**Comment by deniszh** [comment]
Thanks, looks good, but maybe it will be good to have another pair of eye co check it.
/cc @graphite-project/committers

File: whisper.py [hunk 1]
```diff
@@ -286,45 +297,57 @@ def __readHeader(fh):
...
-      raise CorruptWhisperFile("Unable to read header", fh.name)
+    __writeHeaderMetadata(fh, info['aggregationMethod'], info['maxRetention'],
+                          xFilesFactor, len(info['archives']))
 
-    try:
-      newAggregationType = struct.pack(longFormat, aggregationMethodToType[aggregationMethod])
-    except KeyError:
-      raise InvalidAggregationMethod("Unrecognized aggregation method: %s" %
-            aggregationMethod)
+    if AUTOFLUSH:
```
Comment (iksaif) [inline whisper.py]: Looks like your copy-pasting most of this code from somewhere else. Maybe you could move that in a function ?

File: whisper.py [hunk 2]
```diff
@@ -333,7 +356,37 @@ def setAggregationMethod(path, aggregationMethod, xFilesFactor=None):
...
+    raise InvalidAggregationMethod("Unrecognized aggregation method: %s" %
+                                   aggregationMethod)
+
+  try:
+    xFilesFactor = float(xFilesFactor)
+  except:
+    raise InvalidXFilesFactor("Invalid xFilesFactor %s, not a float" %
+                              xFilesFactor)
+
+  if not 0 <= xFilesFactor <= 1:
```
Comment (iksaif) [inline whisper.py]: This looks more readable to me: `if xFilesFactor < 0 or xFilesFactor > 1`

File: whisper.py (hunk 1 above)
Comment (piotr1212) [inline whisper.py]: setXFilesFactor and setAggregationMethod are basically copy+paste. I wasn't sure how to do this without changing the api and keeping it consistent.

I've got this now, still not sure if this is the right way:
https://github.com/piotr1212/whisper/blob/set_xff2/whisper.py#L300-L348
any suggestions?

File: whisper.py (hunk 2 above)
Comment (piotr1212) [inline whisper.py]: will change this

File: whisper.py (hunk 1 above)
Comment (iksaif) [inline whisper.py]: looks cleaner yep

**Comment by piotr1212** [comment]
Are my changes good enough or do I need to refactor more?

**Comment by iksaif** [comment]
LGTM

**Comment by piotr1212** [comment]
cool thanks

**Pull request merged**